runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5291,17 +5291,18 @@ class INTERPRETER_CLASS
 		return rc;
 	}
 
-	/* jdk.internal.vm.Continuation: private static native boolean yieldImpl(); */
+	/* jdk.internal.vm.Continuation: private static native boolean yieldImpl(boolean isFinished); */
 	VMINLINE VM_BytecodeAction
 	yieldContinuationImpl(REGISTER_ARGS_LIST)
 	{
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+		BOOLEAN isFinished = *(U_8*)_sp;
```

Comment:
not expert, but '(I_8)*(I_32*)_sp;' might be more correct (relevant for endianness) 

@babsingh please confirm

Reply:
For boolean Java fields, other places in BytecodeInterpreter use:
```suggestion
		I_32 isFinished = *(I_32*)_sp;
```

Reply:
OK, we must read 32bit, but in native I'd prefer to continue to use some sort of a boolean type (and an explicit cast), not an integer.

Reply:
> perhaps maybe even explicit cast to BOOLEAN too!?

BOOLEAN is defined differently across platforms. I would avoid the cast in case there is data loss (from signed to unsigned).

Reply:
Safer approach to avoid loss of data when `BOOLEAN` is defined as `uint32_t`:
```suggestion
		BOOLEAN isFinished = (0 == *(I_32*)_sp) ? FALSE : TRUE;
```

Reply:
agree, this is safer